src/Fantomas/CodePrinter.fs

```diff
@@ -2030,7 +2030,13 @@ and genExpr astContext synExpr ctx =
                 +> sepNln
                 +> genExpr astContext e
                 +> unindent
-                +> kw WITH !+~ "with"
+                +> (fun ctx ->
+                    let lookupRange =
+                        ctx.MkRangeWith
```

Comment:
This check is a bit too vague I'm afraid.
It will fail if the with keyword is higher up:
```fsharp
[<Test>]
let ``comment is still lost`` () =
    formatSourceString
        false
        """
try
    a
// comment
with


| b -> c
"""
        config
    |> prepend newline
    |> should
        equal
        """
try
    a
// comment
with


| b -> c
"""
```

I would add `mWithToLast` to the active pattern:
([parser source](https://github.com/dotnet/fsharp/blob/db2c9da8d1e76d11217d6da53a64253fd0df0246/src/fsharp/pars.fsy#L3520-L3528))

```fsharp
let (|TryWith|_|) =
    function
    | SynExpr.TryWith (e, _, cs, mWithToLast, _, _, _) -> Some(e, mWithToLast, cs)
    | _ -> None
```

and extract the with range as:
```fsharp
| TryWith (e, mWithToLast, cs) ->
 ....
                    let lookupRange =
                        ctx.MkRangeWith
                            (mWithToLast.StartLine, mWithToLast.StartColumn)
                            (mWithToLast.StartLine, mWithToLast.StartColumn + 4)
```

Reply:
I updated the code with your suggestion. Thank you for the help :)

Reply:
I had to miss your suggested code :/ I'm wondering why this code works even with `+3` like `(mWithToLast.StartLine, mWithToLast.StartColumn + 3)`

Reply:
Thanks for explanation! It really helps. But I must admit for me it was too much rush. I have to slowly finish fantomas youtube videos and look again at the source code and assembly all those puzzles together. I think I intuitively understand what's going on, but I need better understanding... TBH this is my first contact with compiler source code especially in F# (which I think basics I know very well but still I'm wrapping my head around some F# features/patterns which are not common in OOP)

Reply:
Yes, it took me a lot of time to understand this project when I first started out.
That is why I made these videos, to speed up this process for newcomers.
But I'm well aware not everything immediately clicks. Especially all those custom operators.
At first you sorta conceptually understand what they do, and later you really start understanding the events they add to the context.
I'm in the same boot that before this project, I never saw a compiler up close.
That was also a very overwhelming brave new world, but with all things, eventually, you get the hang of it.

Thanks again for all your patience and for engaging in conversation.